mlflow/pipelines/utils/step.py

```diff
@@ -116,8 +116,20 @@ def get_pandas_data_profile(data_frame, title: str):
     max_cols = min(data_frame.columns.size, _MAX_PROFILE_COL_SIZE)
     max_rows = min(max(max_cells // max_cols, 1), _MAX_PROFILE_ROW_SIZE)
     truncated_df = data_frame.drop(columns=data_frame.columns[max_cols:]).sample(
-        n=max_rows, ignore_index=True, random_state=42
+        n=max_rows, ignore_index=True, random_state=42, replace=True
```

Comment:
I don't think we should use `replace=True`; this seems like it could sample the same row twice. See https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.sample.html.